modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/new-js/reducers/fieldSetReducer.js

```diff
@@ -136,6 +131,80 @@ export default (state, action, config) => {
 				parentFieldName,
 			});
 		}
+		case EVENT_TYPES.FIELD_SET.UPDATE: {
+			const {fieldSet} = action.payload;
+			const {fieldTypes} = config;
+			const {editingLanguageId, pages} = state;
+
+			const newPages = pages.map((page) => {
+				const rows = page.rows.map((row) => {
+					const columns = row.columns.map((column) => {
+						const fields = column.fields.reduce((fields, field) => {
```

Comment:
Why not use `PageVisitor` here? Since it does the same thing and we can reduce the noise of nested maps a little bit, also because it covers the use case of nested fields.